lib/sycamore/sycamore/transforms/merge_elements.py

```diff
@@ -256,30 +256,43 @@ def merge(self, elt1: Element, elt2: Element) -> Element:
             new_elt.binary_representation = elt1.binary_representation + elt2.binary_representation
 
         # Merge text representations by concatenation with a newline
-        if elt1.text_representation is None or elt2.text_representation is None:
+        if elt1.type != "Image" and (elt1.text_representation is None or elt2.text_representation is None):
             new_elt.text_representation = elt1.text_representation or elt2.text_representation
             new_elt.data["token_count"] = max(tok1, tok2)
 
         else:
             if new_elt.type == "Image+Text":
                 # text rep = summary(image) + text
-                if "summary" in elt1.properties:
-                    new_elt.text_representation = elt1.properties["summary"] + "\n" + elt2.text_representation
+                if elt1.type == "Image":
+                    if "summary" in elt1.properties and "summary" in elt1.properties["summary"]:
+                        new_elt.text_representation = (
+                            elt1.properties["summary"]["summary"] + "\n" + elt2.text_representation
+                        )
+                    else:
+                        new_elt.text_representation = elt2.text_representation
                 else:
-                    new_elt.text_representation = elt1.text_representation + "\n" + elt2.text_representation
+                    if elt1.text_representation and elt2.text_representation:
+                        new_elt.text_representation = elt1.text_representation + "\n" + elt2.text_representation
+                    else:
+                        new_elt.text_representation = elt2.text_representation
                 new_elt.data["token_count"] = tok1 + 1 + tok2
 
             elif new_elt.type == "Section-header+table":
                 # text rep = header text + table html
                 if hasattr(elt2, "table") and elt2.table:
-                    new_elt.text_representation = elt1.text_representation + "\n" + elt2.table.to_html()
-                    new_elt.data["token_count"] = tok1 + 1 + tok2
+                    if elt1.text_representation:
+                        new_elt.text_representation = elt1.text_representation + "\n" + elt2.table.to_html()
+                        new_elt.data["token_count"] = tok1 + 1 + tok2
                 else:
                     new_elt.text_representation = elt1.text_representation
                     new_elt.data["token_count"] = tok1
             else:
                 # text + text
-                new_elt.text_representation = elt1.text_representation + "\n" + elt2.text_representation
+                new_elt_text_representation = "\n".join(
+                    filter(None, [elt1.text_representation, elt2.text_representation])
```

Comment:
what is the purpose of this? To filter out the empty text_representation for either elt1 or elt2?

Reply:
yes, that's right